src/index.ts

```diff
@@ -305,26 +311,42 @@ export const createClaimWithChildInstance = (
       );
       return tx;
     },
-    getClaimingIDsForUser: async (walletID: string): Promise<string[]> => {
+    getClaimingIDsForUser: async (
+      walletID: string
+    ): Promise<IDWithClaimStatus[]> => {
       const claimingToken = await getClaimingToken(
         config.web3Provider,
         config.claimingRegistrarAddress
       );
-      let childDomain: ethers.BigNumber;
-      const claimingIDs: string[] = [];
+      const IDs: IDWithClaimStatus[] = [];
       const bigNumDomainsOwned = await claimingToken.balanceOf(walletID);
       const numDomainsOwned = bigNumDomainsOwned.toNumber();
+      const promises: Promise<void>[] = [];
       if (numDomainsOwned != 0) {
-        let i = 0;
-        while (i < numDomainsOwned) {
-          childDomain = await claimingToken.tokenOfOwnerByIndex(walletID, i);
-          if (await instance.canBeClaimed(childDomain.toHexString())) {
-            claimingIDs.push(childDomain.toHexString());
-          }
-          i++;
+        let count = 0;
+        while (count < numDomainsOwned) {
+          const getDomain = async () => {
+            const childDomain = await claimingToken.tokenOfOwnerByIndex(
+              walletID,
+              count
+            );
+            const idWithClaimStatus: IDWithClaimStatus = {
+              id: childDomain.toHexString(),
+              canBeClaimed: await instance.canBeClaimed(
+                childDomain.toHexString()
+              ),
+            };
+            IDs.push(idWithClaimStatus);
```

Comment:
This is generally not a good idea to do in promises. Instead you should `return idWithClaimStatus`
Then when you do `await Promise.all(promises)` that will return to you a list of `idWithClaimStatus`